### PR TITLE
Stop using SIGDescribe for upgrade tests to reflect reality

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/98326

```release-note
NONE
```

/kind cleanup
/priority important-longerm

### CODE REVIEW

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Comment by liggitt** [comment]
it looks like some of the upgrade tests are still failing with "You may only call BeforeEach from within a Describe, Context or When" because of this change

opened https://github.com/kubernetes/kubernetes/issues/103697

File: test/e2e/cloud/gcp/cluster_upgrade.go [hunk 1]
```diff
@@ -317,18 +309,15 @@ var _ = SIGDescribe("gpu Upgrade [Feature:GPUUpgrade]", func() {
...
 })
 
-// TODO(#98326): Figure out the right SIG and use appropriate SIGDescribe.
+// TODO(#98326): Move the test to sig-aps dir and use SIGDescribe.
 var _ = ginkgo.Describe("[sig-apps] stateful Upgrade [Feature:StatefulUpgrade]", func() {
 	f := framework.NewDefaultFramework("stateful-upgrade")
 
-	// Create the frameworks here because we can only create them
-	// in a "Describe".
-	testFrameworks := createUpgradeFrameworks(statefulsetUpgradeTests)
```
Comment (spiffxp) [inline test/e2e/cloud/gcp/cluster_upgrade.go:331]: @wojtek-t maybe this comment means these framework creations should be put back?

Comment (wojtek-t) [reply test/e2e/cloud/gcp/cluster_upgrade.go:331]: Those were already put back in subsequent PRs.

You can take a look at this comment:
https://github.com/kubernetes/kubernetes/issues/103697#issuecomment-880148377

**Pull request merged**